### PR TITLE
CHECKOUT-6899 Move method-specific interfaces out of payment integration API package

### PR DESCRIPTION
**Pull request by davidchin**
## What?
Move method-specific interfaces (e.g.: `IdealCustomFormFieldsetValues`, `SepaCustomFormFieldsetValues` etc...) out of payment integration API package

## Why?
They should belong to the method-specific integration package rather than the shared API package.

## Testing / Proof
CircleCI

@bigcommerce/checkout

### CODE REVIEW

File: packages/core/src/app/payment/paymentMethod/PaymentMethodTitle.tsx [hunk 1]
```diff
@@ -211,13 +213,13 @@ const PaymentMethodTitle: FunctionComponent<PaymentMethodTitleProps & WithLangua
     const getSelectedCardType = () => {
         if (!isSelected) {
             return;
-        }
+        }
```
Comment (animesh1987) [inline packages/core/src/app/payment/paymentMethod/PaymentMethodTitle.tsx:216]: Is there extra spacing here?

File: packages/core/src/app/payment/paymentMethod/HostedCreditCardFieldsetValues.tsx [hunk 2]
```diff
@@ -0,0 +1,34 @@
+import { CardInstrumentFieldsetValues } from "../storedInstrument";
```
Comment (animesh1987) [inline packages/core/src/app/payment/paymentMethod/HostedCreditCardFieldsetValues.tsx:1]: Just out of curiosity does this need to be a `.tsx` file?

**Comment by animesh1987** [comment]
Will we be adding a generator/executor in future to lift the payment method specific interfaces, if they are declared in payment method packages?

File: packages/core/src/app/payment/paymentMethod/HostedCreditCardFieldsetValues.tsx (hunk 2 above)
Comment (davidchin) [reply packages/core/src/app/payment/paymentMethod/HostedCreditCardFieldsetValues.tsx]: No reason. I just updated the file extension to be `.ts`

**Review by davidchin — COMMENTED** [review]
> Will we be adding a generator/executor in future to lift the payment method specific interfaces, if they are declared in payment method packages?

No, there isn't because there's no need. Therefore, they could just live inside the payment method packages.

**Review by bc-peng — APPROVED** [review]

**Review by animesh1987 — APPROVED** [review]

**Pull request merged**